esx_service/cli/vmdkops_admin.py

```diff
@@ -726,38 +731,382 @@ def get_version():
+        return error_info, None
+    
+    error_info, tenant = auth._auth_mgr.get_tenant(name)
+    return error_info, tenant
+
+def create_tenant_in_db(name, description, default_datastore, default_privileges, vms, privileges):
+    try:
+        auth.connect_auth_db()
+    except auth_data.DbConnectionError, e:
+        error_info = "Failed to connect auth DB({0})".format(e)
+        return error_info, None
+
+    error_info, tenant = auth._auth_mgr.create_tenant(name = name, 
+                                                      description = description, 
+                                                      default_datastore = default_datastore, 
+                                                      default_privileges = default_privileges, 
+                                                      vms = vms, 
+                                                      privileges = privileges)
+    return error_info, tenant
+
+def get_tenant_list_from_db():
+    try: 
+        auth.connect_auth_db()
+        error_info, tenant_list = auth._auth_mgr.list_tenants()
+    except auth_data.DbConnectionError, e:
+        error_info = "Failed to connect auth DB({0})".format(e)
+        return error_info, None
+
+    error_info, tenant_list = auth._auth_mgr.list_tenants()
+    return error_info, tenant_list
+
+def operation_fail(error_info):
+    print error_info
+    return error_info
+
+def tenant_ls_headers():
+    """ Return column names for tenant ls command """
+    headers = ['Uuid', 'Name', 'Description', 'Default_datastore', 'VM_list']
+    return headers
+
+def generate_vm_list(vms):
+    """ Generate a comma separated string with given list """
+    res = ""
+    for vm in vms:
+        # vm[1] is vm_name, vm has format (vm_uuid, vm_name)
+        res = res + vm[1]
+        res = res + ","
+        
+    if res:
+        res = res[:-1]
+
+    return res    
+
+def generate_tenant_ls_rows(tenant_list):
+    """ Generate output for tenant ls command """
+    rows = []
+    for tenant in tenant_list:
+        uuid = tenant.id
+        name = tenant.name
+        description = tenant.description
+        default_datastore = tenant.default_datastore
+        vm_list = generate_vm_list(tenant.vms)
+        rows.append([uuid, name, description, default_datastore, vm_list])
+    
+    return rows
 
-NOT_IMPLEMENTED = "Not implemented"
+def generate_tuple_from_vm_list(vm_list):
+    """ Generate a list of (vm_uuid, vm_name) pair """
+    vms = []
+    for vm_name in vm_list:
+        vm_uuid = vmdk_utils.get_vm_uuid_by_name(vm_name)
+        if not vm_uuid:
+            error_info =  "Cannot find vm_uuid for vm {0}, tenant_create failed".format(vm_name)
+            return error_info, None
+        vms.append((vm_uuid, vm_name))
+    
+    return None, vms
 
 def tenant_create(args):
-    print(NOT_IMPLEMENTED)
-
+    """ Handle tenant create command """
+    error_info, vms = generate_tuple_from_vm_list(args.vm_list)
```

Comment:
Suggest not persisting VM name any where in the DB. As #652 shows, a VM name can change and the user can get confused if old VM names are popping up in admin CLI commands. The UUID is unique and matches an instance of an existing VM.